controllers/validation_test.go

```diff
@@ -0,0 +1,220 @@
+package controllers
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/tools/record"
+	"net/http"
+	"net/http/httptest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"testing"
+)
+
+func TestClusterGroupUpgradeReconciler_extractOpenshiftImagePlatformFromPolicies(t *testing.T) {
+
+	policy := `---
```

Comment:
these policies should probably all be const no?

Reply:
Most actually. One of them needs to be variable since it has to make fake http call. But changed the rest to const (where there is no https calls)